src/compiler/legend.ts

```diff
@@ -1,44 +1,38 @@
-'use strict';
+import {setter, getter} from '../util';
+import * as time from './time';
+import {COLOR, SHAPE, SIZE} from '../consts';
+import {Q, O, N, T} from '../consts';
 
-require('../globals');
-
-var time = require('./time'),
-  util = require('../util'),
-  setter = util.setter,
-  getter = util.getter;
-
-var legend = module.exports = {};
-
-legend.defs = function(encoding, style) {
+export function defs(encoding, style) {
   var defs = [];
 
   if (encoding.has(COLOR) && encoding.encDef(COLOR).legend) {
-    defs.push(legend.def(COLOR, encoding, {
+    defs.push(def(COLOR, encoding, {
       fill: COLOR
     }, style));
   }
 
   if (encoding.has(SIZE) && encoding.encDef(SIZE).legend) {
-    defs.push(legend.def(SIZE, encoding, {
+    defs.push(def(SIZE, encoding, {
       size: SIZE
     }, style));
   }
 
   if (encoding.has(SHAPE) && encoding.encDef(SHAPE).legend) {
-    defs.push(legend.def(SHAPE, encoding, {
+    defs.push(def(SHAPE, encoding, {
       shape: SHAPE
     }, style));
   }
   return defs;
 };
 
-legend.def = function(name, encoding, def, style) {
+export function def(name, encoding, def, style) {
   var timeUnit = encoding.encDef(name).timeUnit;
 
-  def.title = legend.title(name, encoding);
+  def.title = title(name, encoding);
   def.orient = encoding.encDef(name).legend.orient;
 
-  def = legend.style(name, encoding, def, style);
+  def = exports.style(name, encoding, def, style);
```

Comment:
I don't think renaming to `vlStyle` will work here ... 

Otherwise, you get `legend.vlStyle` which is kinda weird.  

But assuming we remove `style` parameter, this should be fine